dali/pipeline/operators/color/hsv.h

```diff
@@ -0,0 +1,212 @@
+#include "dali/pipeline/data/views.h"
+#include "dali/pipeline/operators/common.h"
+#include "dali/pipeline/operators/operator.h"
+#include "dali/kernels/kernel_manager.h"
+#include "dali/kernels/tensor_shape_print.h"
+#include "dali/kernels/imgproc/pointwise/linear_transformation_cpu.h"
+
+
+namespace dali {
+namespace hsv {
+
+/**
+ * Names of arguments
+ */
+extern const std::string kHue;
+extern const std::string kSaturation;
+extern const std::string kValue;
+extern const std::string kOutputType;
+
+/**
+ * Color space conversion
+ */
+const mat3 Rgb2Yiq = {{
+                              {.299f, .587f, .114f},
+                              {.596f, -.274f, -.321f},
+                              {.211f, -.523f, .311f}
+                      }};
+
+// Inversion of Rgb2Yiq, but pre-calculated, cause mat<> doesn't do inversion.
+const mat3 Yiq2Rgb = {{
+                              {1, .956f, .621f},
+                              {1, -.272f, -.647f},
+                              {1, -1.107f, 1.705f}
+                      }};
+
+
+inline mat3 compose_hue(float hue /* hue hue hue */ ) {
+  const auto h_rad = hue * M_PI / 180;
+  mat3 ret = mat3::eye();  // rotation matrix
+  ret(1, 1) = cos(h_rad);
+  ret(2, 2) = cos(h_rad);
+  ret(1, 2) = -sin(h_rad);
+  ret(2, 1) = sin(h_rad);
+  return ret;
+}
+
+
+inline mat3 compose_saturation(float saturation) {
+  mat3 ret = mat3::eye();
+  ret(1, 1) = saturation;
+  ret(2, 2) = saturation;
+  return ret;
+}
+
+
+inline mat3 compose_value(float value) {
+  return mat3::diag(value);
+}
+
+}  // namespace hsv
+
+
+
+template <typename Backend>
+class Hsv : public Operator<Backend> {
+ public:
+  ~Hsv() override = default;
+
+  DISABLE_COPY_MOVE_ASSIGN(Hsv);
+
+ protected:
+  explicit Hsv(const OpSpec &spec) :
+          Operator<Backend>(spec),
+          output_type_(spec.GetArgument<DALIDataType>(hsv::kOutputType)) {
+    GetSingleOrRepeatedArg(spec, hue_, hsv::kHue, batch_size_);
+    GetSingleOrRepeatedArg(spec, saturation_, hsv::kSaturation, batch_size_);
+    GetSingleOrRepeatedArg(spec, value_, hsv::kValue, batch_size_);
+    if (std::is_same<Backend, GPUBackend>::value) {
+      kernel_manager_.Resize(1, 1);
+    } else {
+      kernel_manager_.Resize(num_threads_, batch_size_);
+    }
+    assert(hue_.size() == batch_size_);
+    assert(saturation_.size() == batch_size_);
+    assert(value_.size() == batch_size_);
+  }
+
+
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+
+  /**
+   * @brief Creates transformation matrices based on given args
+   */
+  std::vector<mat3>
+  determine_transformation(const std::vector<float> &hue, const std::vector<float> &saturation,
+                           const std::vector<float> &value) const {
+    using namespace hsv;  // NOLINT
+    std::vector<mat3> ret;
+
+    auto size = hue.size();
+    ret.resize(size);
+    for (size_t i = 0; i < size; i++) {
+      ret[i] = Yiq2Rgb * compose_hue(hue[i]) * compose_saturation(saturation[i]) *
```

Comment:
Nitpick: I thought you would provide a reference why those matrices are rotations, etc.